Domains: Check for validation errors before showing the privacy modal

klimeryk opened the pull request:
When in the `privacyCheckbox` A/B test in the `checkbox` variation (introduced in #3838), any validation error will _not_ prevent the privacy modal from showing, but it _will_ prevent from the buttons there from doing anything. This leads to a confused user (he can't see the errors because of the modal).
This fixes this issue - but I wonder if the test should not be restarted, because I can imagine this could skew the results of the test _a bit_ :wink:

#### Testing
* Make sure you are in the `checkbox` variation of the `privacyCheckbox` A/B test.
* Add a domain to cart.
* On the Domain Contact form, cause a validation error (or start already with invalid information - like a phone number without the country code from a previous purchase).
* Click the `Continue to checkout` button.
* The modal should _not_ be shown - instead, the first field with the validation error should be focused.
* Only when all the validation errors have been dealt with, clicking the `Continue to checkout` button should bring up the modal.

Of course, please test if there are no regressions in the `original` variation too.

Fixes #5531

@aidvu @umurkontaci - code review
@breezyskies @drewblaisdell @scruffian @Tug from the A/B test PR

drewblaisdell commented:
Thanks for fixing this. I was still able to replicate the issue when [no field had been initialized](https://cloudup.com/cockdGvUQdK), so I added a commit to not display the privacy protection modal until the domain details have been validated.

klimeryk commented:
Thank _you_, Drew, nice catch! :+1: Your fix is better than mine and, in fact, my is not needed anymore :) I've rebased and removed my commit - let me know if it's good to merge.
